Spring-server/src/main/java/ikpi63holding/postclient/ApiController.java

```diff
@@ -0,0 +1,125 @@
+package ikpi63holding.postclient;
+
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+public class ApiController {
+
+    private final UserRepository repository;
+
+    ApiController(UserRepository repository) {
+        this.repository = repository;
+    }
+
+    @GetMapping("/folders/{id}")
+    public List<String> getFolders(@PathVariable Long id) throws Exception {
+        User user = repository.findById(id).orElse(null);
+        ImapConnector connector =
+                new ImapConnector(user.getMailLogin(), user.getMailPassword(), user.getImapAddr());
+        return connector.getFolders();
+    }
+
+    @PostMapping("/mail/{id}")
+    public void sendMail(@PathVariable Long id, @RequestBody PostMessage message) throws Exception {
+        User user = repository.findById(id).orElse(null);
+        SmtpConnector smtpConnector =
+                new SmtpConnector(user.getMailLogin(), user.getMailPassword(), user.getSmtpAddr());
+        smtpConnector.sendEmail(message);
+        ImapConnector imapConnector =
+                new ImapConnector(user.getMailLogin(), user.getMailPassword(), user.getImapAddr());
+        imapConnector.saveSendMessage(message);
+    }
+
+    @GetMapping("/mail/{id}")
+    public List<PostMessage> getMail(@PathVariable Long id,
+            @RequestParam(name = "folder", required = false, defaultValue = "INBOX")
+                    String folderName) throws Exception {
+        User user = repository.findById(id).orElse(null);
+        ImapConnector imapConnector =
+                new ImapConnector(user.getMailLogin(), user.getMailPassword(), user.getImapAddr());
+        return imapConnector.getMailList(folderName);
+    }
+
+    @DeleteMapping("/mail/{id}")
+    public void getMail(@PathVariable Long id, @RequestBody PostMessage message) throws Exception {
```

Comment:
I think we should rename this function "deleteMail"

Reply:
I agree

Reply:
Yea, I was making this mapping for testing, so didn't care about correct naming, should fix